XamarinCommunityToolkit.UnitTests/Converters/ByteArrayToImageSourceConverter_Tests.cs

```diff
@@ -13,20 +13,64 @@ public class ByteArrayToImageSourceConverter_Tests
         public void ByteArrayToImageSourceConverter()
         {
             var byteArray = new byte[] { 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20 };
-            var expectedValue = ImageSource.FromStream(() => new MemoryStream(byteArray));
+
+            var memoryStream = new MemoryStream(byteArray);
+
+            var expectedValue = ImageSource.FromStream(() => memoryStream);
 
             var byteArrayToImageSourceConverter = new ByteArrayToImageSourceConverter();
 
             var result = byteArrayToImageSourceConverter.Convert(byteArray, typeof(ByteArrayToImageSourceConverter), null, CultureInfo.CurrentCulture);
-            Assert.Equal(result, expectedValue);
+
+            Assert.True(StreamEquals(GetStreamFromImageSource((ImageSource)result), memoryStream));
         }
 
         [Theory]
         [InlineData("Random String Value")]
         public void InvalidConverterValuesReturnsNull(object value)
         {
             var byteArrayToImageSourceConverter = new ByteArrayToImageSourceConverter();
-            Assert.Null(byteArrayToImageSourceConverter.Convert(value, typeof(ByteArrayToImageSourceConverter), null, CultureInfo.CurrentCulture));
+
+            Assert.Throws<ArgumentException>(() => byteArrayToImageSourceConverter.Convert(value, typeof(ByteArrayToImageSourceConverter), null, CultureInfo.CurrentCulture));
+        }
+
+        Stream GetStreamFromImageSource(ImageSource imageSource)
+        {
+            var streamImageSource = (StreamImageSource)imageSource;
+
+            var cancellationToken = System.Threading.CancellationToken.None;
+            var task = streamImageSource.Stream(cancellationToken);
+            return task.Result;
+        }
+
+        bool StreamEquals(Stream a, Stream b)
+        {
+            if (a == b)
+            {
+                return true;
+            }
+
+            if (a == null || b == null)
+            {
+                throw new ArgumentNullException(a == null ? "a" : "b");
+            }
+
+            if (a.Length != b.Length)
+            {
+                return false;
+            }
+
+            for (var i = 0; i < a.Length; i++)
+            {
+                var aByte = a.ReadByte();
+                var bByte = b.ReadByte();
+                if (aByte.CompareTo(bByte) != 0)
+                {
+                    return false;
+                }
+            }
+
+            return true;
```

Comment:
```suggestion
            if (a == b)
                return true;
                
            if (a == null ||
                b == null ||
                a.Length != b.Length)
                return false;
                
            for (var i = 0; i < a.Length; i++)
                if (a.ReadByte() != b.ReadByte())
                    return false;
                    
            return true;
```

Reply:
I don't think we should raise ArgumentNullException if a == null || b == null
(btw we don't raise if both equal null cuz null == null // TRUE)